### PR TITLE
Modernize policy attachments

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR:
* Makes a previously-attached policy an inline policy (https://github.com/cisagov/cool-pca-iam/commit/5f2635c8ce30fa58da2765c19355a7145b9938af)
* Attaches a policy to the provisioners group that was previously attached elsewhere (https://github.com/cisagov/cool-pca-iam/commit/e154b2a7352c27631bd5b14bb4b22933ad3eb367)
* Cleans up some Terraform syntax (https://github.com/cisagov/cool-pca-iam/commit/db4452d54004ff61fb009796d18c26c8ab494e48)

## 💭 Motivation and context ##

The policy that was switched to inline is one that allows assumption of the `ProvisionPrivateDNSRecords` role in the Shared Services account.  It was changed to inline because other repos (e.g. [`cisagov/cool-domain-manager-iam`](https://github.com/cisagov/cool-domain-manager-iam)) also require the exact same policy, however we do not currently have a good centralized location where this policy can be Terraformed.  So to avoid multiple repos attempting to create the same policy with the same name (which wouldn't work), the decision was made to simply make this an inline policy for now.

The policy that was attached to the provisioner's group is the one that allows assumption of the role in the Terraform account that can read the cool-sharedservices-networking Terraform state.  This policy was previously attached via another repo, but that was changed recently.  Refer to https://github.com/cisagov/cool-pca-transitgateway-attachment/pull/5 and https://github.com/cisagov/cool-sharedservices-networking/pull/41 for details.

## 🧪 Testing ##

This change was successfully Terraformed and all pre-commit hooks pass.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

### CODE REVIEW

File: group_membership.tf [hunk 1]
```diff
@@ -2,7 +2,7 @@
 resource "aws_iam_user_group_membership" "provisioner" {
   provider = aws.users
 
-  for_each = { for username, attributes in var.users : username => "" if contains(attributes["roles"], "provisioner") }
+  for_each = toset([for username, attributes in var.users : username if contains(attributes["roles"], "provisioner")])
```
Comment (jsf9k) [inline group_membership.tf:5]: Hallelujah!

**Review by jsf9k — APPROVED** [review]

File: group_membership.tf (hunk 1 above)
Comment (dav3r) [reply group_membership.tf:5]: Ha- there's one place to make this change, over in [`cool-auditor-iam`](https://github.com/cisagov/cool-auditor-iam/blob/develop/group_membership.tf) - I'll get to it soon.

**Review by hillaryj — APPROVED** [review]
🔋

**Review by mcdonnnj — APPROVED** [review]
Does the things ✔

**Pull request merged**